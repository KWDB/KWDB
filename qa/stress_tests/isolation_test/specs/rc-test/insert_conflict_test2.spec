setup
{
  CREATE TABLE ints (key int primary key, val text);
}

teardown
{
  DROP TABLE ints;
}

session s1
setup         { BEGIN transaction isolation level read committed; }
step s1w      { INSERT INTO ints(key, val) VALUES(1, 'donothing1') ON CONFLICT DO NOTHING; }
step s1c      { COMMIT; }
step s1a      { ABORT; }

session s2
setup         { BEGIN transaction isolation level read committed; }
step s2w      { INSERT INTO ints(key, val) VALUES(1, 'donothing2') ON CONFLICT DO NOTHING; }
step s2r      { SELECT * FROM ints; }
step s2c      { COMMIT; }


permutation s1w s2w s1c s2r s2c
permutation s1w s2w s1a s2r s2c
