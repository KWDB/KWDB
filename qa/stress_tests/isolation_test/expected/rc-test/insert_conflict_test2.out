Parsed test spec with 2 sessions

starting permutation: s1w s2w s1c s2r s2c
step s1w: INSERT INTO ints(key, val) VALUES(1, 'donothing1') ON CONFLICT DO NOTHING;  finished.
step s2w: INSERT INTO ints(key, val) VALUES(1, 'donothing2') ON CONFLICT DO NOTHING; <waiting ...>
step s1c: COMMIT;  finished.
step s2w: <... completed>
ERROR:  flow: memory budget exceeded: 10240 bytes requested, 0 currently allocated, 0 bytes in budget
step s2r: SELECT * FROM ints;  finished.
ERROR:  current transaction is aborted, commands ignored until end of transaction block
step s2c: COMMIT;  finished.

starting permutation: s1w s2w s1a s2r s2c
step s1w: INSERT INTO ints(key, val) VALUES(1, 'donothing1') ON CONFLICT DO NOTHING;  finished.
step s2w: INSERT INTO ints(key, val) VALUES(1, 'donothing2') ON CONFLICT DO NOTHING; <waiting ...>
step s1a: ABORT;  finished.
step s2w: <... completed>
step s2r: SELECT * FROM ints;  finished.
key|val       
---+----------
  1|donothing2
(1 row)

step s2c: COMMIT;  finished.
